### ISSUE TITLE
dynamic_k_matching question?

### ISSUE THREAD

**Issue by DacDinh147** [Megvii-BaseDetection/YOLOX#787]
I see in your code https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/models/yolo_head.py#L632
```
def dynamic_k_matching(self, cost, pair_wise_ious, gt_classes, num_gt, fg_mask):
        # Dynamic K
        # ---------------------------------------------------------------
        matching_matrix = torch.zeros_like(cost)

        ...
        fg_mask[fg_mask.clone()] = fg_mask_inboxes  # line 632 in file

       ...
        return num_fg, gt_matched_classes, pred_ious_this_matching, matched_gt_inds
```
as I understand, modified foreground mask `fg_mask` only exists in the scope of this function without return its value as output, and the returned value in function `get_assignments` https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/models/yolo_head.py#L516 will not reflect this change. Since you are trying to refine positive anchor again by low cost anchor matching mask, I think this may be a import bug or may be I am wrong. So, I set this it a question.

**Comment by DacDinh147** [comment]
Oh, sorry, it actually change the value. I close the question.
